### ISSUE TITLE
prepare scripts ipc.remoteException(NullPointerException)

### ISSUE THREAD

**Issue by vishalmehta1991** [intel-hadoop/HiBench#118]
I have this exception with all prepare scripts except for sleep,wordcount,sort and terasort.

15/07/24 12:10:58 INFO mapred.FileInputFormat: Total input paths to process : 1
15/07/24 12:10:58 INFO mapreduce.JobSubmitter: Cleaning up the staging area /tmp/hadoop-yarn/staging/vmehta/.staging/job_1437732511086_0001
Exception in thread "main" org.apache.hadoop.ipc.RemoteException(java.lang.NullPointerException): java.lang.NullPointerException
        at org.apache.hadoop.hdfs.server.blockmanagement.DatanodeManager.sortLocatedBlocks(DatanodeManager.java:359)
        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getBlockLocations(FSNamesystem.java:1787)
        at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.getBlockLocations(NameNodeRpcServer.java:542)
        at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.getBlockLocations(ClientNamenodeProtocolServerSideTranslatorPB.java:362)
        at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
        at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:619)
        at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:962)
        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2039)
        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2035)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:415)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1628)
        at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2033)

**Comment by GayathriMurali** [comment]
Can you provide more information on your setup? Number of nodes, Hadoop,Java and Spark versions? Also please share the log information on this error.

**Comment by vishalmehta1991** [comment]
Hello,
I found the bug, the network topology script was not present but still the topology property was defined by previous installation. Except sleep,wordcount,sort and terasort (mapreduce based) all other workloads were getting nullpointer exception. Once the property was commented in core-site.xml the exception was removed. This was supposed to be a bug in Apache hadoop-2.0 and has been closed in 2.2. I am using hadoop 2.6, but looks like only hadoop-examples.jar has handled this bug, as the workloads from them do work.  

Thanks for support though.
